src/kademlia/dht_storage.cpp

```diff
@@ -0,0 +1,355 @@
+				dht_immutable_table_t::iterator j = std::min_element(m_immutable_table.begin()
+						, m_immutable_table.end()
+						, immutable_item_comparator(m_id));
+
+				TORRENT_ASSERT(j != m_immutable_table.end());
+				free(j->second.item.value);
+				m_immutable_table.erase(j);
+				m_counters.inc_stats_counter(counters::dht_immutable_data, -1);
+			}
+
+			boost::tie(i, boost::tuples::ignore) = m_immutable_table.insert(
+					std::make_pair(target, to_add));
+			m_counters.inc_stats_counter(counters::dht_immutable_data);
+		}
+
+		touch_item(&i->second.popularity, addr);
+	}
+
+	bool dht_default_storage::get_mutable_item(sha1_hash const& target
+		, dht_mutable_item& item) const
+	{
+		dht_mutable_table_t::const_iterator i = m_mutable_table.find(target);
+		if (i == m_mutable_table.end()) return false;
+
+		item = i->second.item;
+
+		return true;
+	}
+
+	void dht_default_storage::put_mutable_item(sha1_hash const& target, dht_mutable_item const& to_add, address addr)
+	{
+		dht_mutable_table_t::iterator i = m_mutable_table.find(target);
+		if (i == m_mutable_table.end())
+		{
+			// this is the case where we don't have an item in this slot
+			// make sure we don't add too many items
+			if (int(m_mutable_table.size()) >= m_settings.max_dht_items)
+			{
+				// delete the least important one (i.e. the one
+				// the fewest peers are announcing)
+				dht_mutable_table_t::iterator j = std::min_element(m_mutable_table.begin()
+						, m_mutable_table.end()
+						, boost::bind(&dht_popularity_item::num_announcers
+							, boost::bind(&dht_mutable_entry::popularity
+								, boost::bind(&dht_mutable_table_t::value_type::second, _1))));
+				TORRENT_ASSERT(j != m_mutable_table.end());
+				free(j->second.item.value);
+				free(j->second.item.salt);
+				m_mutable_table.erase(j);
+				m_counters.inc_stats_counter(counters::dht_mutable_data, -1);
+			}
+
+			boost::tie(i, boost::tuples::ignore) = m_mutable_table.insert(
+					std::make_pair(target, to_add));
+			m_counters.inc_stats_counter(counters::dht_mutable_data);
+		}
+		else
+		{
+			boost::tie(i, boost::tuples::ignore) = m_mutable_table.insert(
+					std::make_pair(target, to_add));
+		}
+
+		touch_item(&i->second.popularity, addr);
+	}
+
+	void dht_default_storage::tick()
+	{
+		time_point now(aux::time_now());
+
+		for (dht_immutable_table_t::iterator i = m_immutable_table.begin();
+			 i != m_immutable_table.end();)
+		{
+			if (i->second.popularity.last_seen + minutes(60) > now)
+			{
+				++i;
+				continue;
+			}
+			free(i->second.item.value);
+			m_immutable_table.erase(i++);
+			m_counters.inc_stats_counter(counters::dht_immutable_data, -1);
+		}
+
+		// look through all peers and see if any have timed out
+		for (table_t::iterator i = m_map.begin(), end(m_map.end()); i != end;)
+		{
+			torrent_entry& t = i->second;
+			node_id const& key = i->first;
+			++i;
+			purge_peers(t.peers);
+
+			// if there are no more peers, remove the entry altogether
+			if (t.peers.empty())
```

Comment:
personally I prefer the style of: if (!t.peers.empty()) continue;

that way you save an indentation step below